NuRadioReco/examples/PhasedArray/SNR_curves/T02RunSNR_RNOG.py

```diff
@@ -0,0 +1,280 @@
+                channel.set_trace(trace, sampling_rate=new_sampling_rate)
+
+            # Adding noise AFTER the SNR calculation
+            channelGenericNoiseAdder.run(self._evt, self._station, self._det, amplitude = 1.0 / Vrms_ratio,
+                                         min_freq=min_freq, max_freq=max_freq, type='rayleigh')
+
+            # bandpass filter trace, the upper bound is higher then the sampling rate which makes it just a highpass filter
+            channelBandPassFilter.run(self._evt, self._station, self._det, passband=[0 * units.MHz, 240.0 * units.MHz],
+                                      filter_type='cheby1', order=9, rp=.1)
+            channelBandPassFilter.run(self._evt, self._station, self._det, passband=[80.0 * units.MHz, 230.0 * units.MHz],
+                                      filter_type='cheby1', order=4, rp=.1)
+
+            if(phase):
+                has_triggered = triggerSimulator.run(self._evt, self._station, self._det,
+                                                     Vrms = 1.0,
+                                                     threshold = threshold,
+                                                     triggered_channels=channels,
+                                                     phasing_angles=phasing_angles,
+                                                     ref_index = 1.75,
+                                                     trigger_name='primary_phasing',
+                                                     trigger_adc=False, # Don't have a seperate ADC for the trigger
+                                                     adc_output='voltage', # output in volts
+                                                     trigger_filter=None,
+                                                     upsampling_factor=upsampling_factor,
+                                                     window=int(32 * new_sampling_rate * upsampling_factor / 2),
+                                                     step = int(16 * new_sampling_rate * upsampling_factor / 2))
+            else:
+                original_traces_ = self._station.get_channel(4).get_trace()
+
+                squared_mean, num_frames = triggerSimulator.powerSum(coh_sum=original_traces_,
+                                                                     window=int(32 * new_sampling_rate / 2),
+                                                                     step=int(16 * new_sampling_rate / 2),
+                                                                     adc_output=f'voltage')
+
+                squared_mean_threshold = np.power(threshold, 2.0)
+
+                has_triggered = (True in (squared_mean > squared_mean_threshold))
+
+            if(has_triggered):
+                print('Trigger on noise... ')
+
+        filtered_noise_traces = {}        
+        for channel in self._station.iter_channels():
+            filtered_noise_traces[channel.get_id()] = channel.get_trace()            
```

Comment:
doesn't this set `filtered_noise_traces` to the noise fluctuation that triggered the phased array where the Askaryan signal is zero ?

Reply:
This sets filtered_noise_traces to the noise fluctuation that didn't trigger the phased array, and then the filtered askaryan signal gets added back in later on.